AK/FloatingPointStringConversions.cpp

```diff
@@ -1503,34 +1489,34 @@ class MinimalBigInt {
         // Top word should be non-zero
         VERIFY(m_words[m_used_length - 1] != 0);
 
-        auto leading_zeros = count_leading_zeroes(m_words[m_used_length - 1]);
-        if (m_used_length == 1)
-            return m_words[0] << leading_zeros;
+        auto top_u64_start = static_cast<size_t>(size_in_bits()) - 64;
+        u64 top_u64 = 0;
 
-        for (size_t i = 0; i < m_used_length - 2; i++) {
-            if (m_words[i] != 0) {
-                has_truncated_bits = true;
-                break;
-            }
-        }
+        for (size_t i = 0; i < m_used_length; ++i) {
+            size_t word_start = i * word_size, word_end = word_start + word_size;
```

Comment:
(Style nitpick) I think we usually never have multiple variable declarations on the same line.
I initially thought this was some complicated expression.